### PR TITLE
feat(gcli): log program ID after uploading program

### LINKED ISSUE #3549: Logging precise transaction hash in gsdk
### Problem to Solve

```
cli upload ./target/wasm32-unknown-unknown/release/dummy_wasm.opt.wasm
[INFO ] Run tx: Gear::upload_program
[INFO ] Submitted extrinsic Gear::upload_program
[INFO ]         Status: Validated
[INFO ]         Status: Broadcast( 15 )
[INFO ]         Status: InBestBlock( block hash: 0x2a9c…e508, extrinsic hash: 0x16c3…1774 )
[INFO ]         Status: Finalized( block hash: 0x2a9c…e508, extrinsic hash: 0x16c3…1774 )
[INFO ] Successfully submitted call Gear::upload_program 0x16c3…1774 at 0x2a9c…e508!
[INFO ]         Balance spent: 61804266300
```

### Possible Solution

the transaction hash is trimmed now on printing, perform a prettier format for this logging btw

### Notes

_No response_

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #3549 
Ref #3548 

- [x] log precise block and transaction hash
- [x] log program ID after uploading program
- [x] log message ID after sending message

### upload program

```
10:54PM ./target/release/gcli --endpoint ws://localhost:9944 upload target/wasm32-unknown-unknown/debug/demo_messager.opt.wasm 
[INFO ] Connecting to ws://localhost:9944 ...
[INFO ] Pending Gear::upload_program ...
[INFO ]         Status: Validated
[INFO ]         Status: InBestBlock( block hash: 0x8f0f…d31e, extrinsic hash: 0xf95b…5f6e )
[INFO ]         Status: Finalized( block hash: 0x8f0f…d31e, extrinsic hash: 0xf95b…5f6e )
[INFO ] Submitted Gear::upload_program !
[INFO ]         Block Hash: 0x8f0f897fb2b507acadc108093f5c246c9a59197cdbdc674e5e2c309909bfd31e
[INFO ]         Transaction Hash: 0xf95b3892e4a86fcf22e63ab2405a2bda46197a0fbdc2f00e1a0a2e2799d35f6e
[INFO ]         Balance spent: 114843081325
[INFO ] Code ID: 0x7ca77d2d9e6b61614d825b2f7f10d44e5fff11e896ee91fe757e7201d6e0a808
[INFO ] Program ID: 0x676165415568fb84e9608b38a5d781ef001f8be438d11c7899e7445203481582
[INFO ] Init Message ID: 0x5d4bb1815ee31b74af82598da8796fa57cdcca3d3ea803c4d251a918b3662aa3
```

### send message

```
 6:14AM ./target/release/gcli --endpoint ws://localhost:9944 send 0xf50340c387a87a9d64f4ed9a949857c78a3cf320ad814cd0bcb47a66346f0c39 
[INFO ] Pending Gear::send_message ...
[INFO ]         Status: Validated
[INFO ]         Status: InBestBlock( block hash: 0xedfd…366d, extrinsic hash: 0xb345…64f1 )
[INFO ]         Status: Finalized( block hash: 0xedfd…366d, extrinsic hash: 0xb345…64f1 )
[INFO ] Submitted Gear::send_message !
[INFO ]         Block Hash: 0xedfdef01265973ca67b8c12031e71a004fd782ede197df1cc60c2c386481366d
[INFO ]         Transaction Hash: 0xb345d512672b5fd92f6b1a5e12d5b1b8db330d43b2388445ad2d34c00bf764f1
[INFO ]         Balance spent: 35581758400
[INFO ] Message ID: 0x5bc2a23d82cbb3da5a2c9ccca5c9bc723b7e89a80119510abc8dbb1f38218f8f
```

@gear-tech/dev

### CODE REVIEW

**Comment by clearloop** [comment]
switch to gclient in this PR

**Review by ukint-vs — APPROVED** [review]

File: gcli/src/cmd/upload.rs [hunk 1]
```diff
@@ -66,11 +66,19 @@ impl Upload {
...
         // Estimate gas and upload program.
         let gas_limit = signer.api().cmp_gas_limit(gas)?;
-        signer
+        let tx = signer
             .calls
             .upload_program(code, self.salt.to_vec()?, payload, gas_limit, self.value)
             .await?;
 
+        for event in signer.api().events_of(&tx).await? {
+            if let Event::Gear(GearEvent::ProgramChanged { id, change }) = event {
```
Comment (breathx) [inline gcli/src/cmd/upload.rs:75]: init message id?

**Review by breathx — APPROVED** [review]

File: gcli/src/cmd/upload.rs [hunk 2]
```diff
@@ -66,11 +66,19 @@ impl Upload {
...
+                log::info!("Program ID: 0x{}", hex::encode(id.0));
+                log::info!("Program status: {:?}", change);
```
Comment (NikVolf) [inline gcli/src/cmd/upload.rs:77]: Code hash ?

Comment (clearloop) [reply gcli/src/cmd/upload.rs]: ef4a2ed

File: gcli/src/cmd/upload.rs (hunk 1 above)
Comment (clearloop) [reply gcli/src/cmd/upload.rs]: ef4a2ed

**Pull request merged**